aderyn_core/src/detect/detector.rs

```diff
@@ -42,6 +49,73 @@ pub fn get_all_detectors() -> Vec<Box<dyn Detector>> {
     ]
 }
 
+pub fn get_all_detectors_names() -> Vec<String> {
+    get_all_detectors().iter().map(|d| d.name()).collect()
+}
+
+// Note to maintainers: DO NOT CHANGE THE ORDER OF THESE DERIVE ATTRIBUTES
+#[derive(Debug, PartialEq, EnumString, Display)]
+#[strum(serialize_all = "kebab-case")]
+pub(crate) enum DetectorNamePool {
+    DelegateCallInLoop,
```

Comment:
Hmmmm, on second thoughts @TilakMaddy , do you think this Enum adds too much back-and-forth between here and detectors?

Reply:
Yes, Alex! But it's when a new detector gets formalized to aderyn codebase - we'd have to create a new entry in the detector pool to return in the `name()` function and the rust compiler will guide us to make the remaining changes.

Maybe not back & forth ...more like back once, forth from then on : ) :P